pkg/kubelet/dockershim/docker_sandbox.go

```diff
@@ -203,28 +203,37 @@ func (ds *dockerService) StopPodSandbox(podSandboxID string) error {
 			errList = append(errList, err)
 		}
 	}
-	if err := ds.client.StopContainer(podSandboxID, defaultSandboxGracePeriod); err != nil {
-		glog.Errorf("Failed to stop sandbox %q: %v", podSandboxID, err)
-		// Do not return error if the container does not exist
-		if !dockertools.IsContainerNotFoundError(err) {
-			errList = append(errList, err)
+
+	// Get all the containers in the PodSandbox.
+	containers, err := ds.getContainersInPodSandbox(podSandboxID)
+	if err != nil {
+		errList = append(errList, err)
+	}
+
+	// Stop all containers in the sandbox.
+	for i := range containers {
+		if containers[i].State == "running" {
+			if err := ds.StopContainer(containers[i].ID, int64(defaultSandboxGracePeriod)); err != nil && !dockertools.IsContainerNotFoundError(err) {
```

Comment:
It's better to add a `sync.WaitGroup{}` wrapper here to stop all containers in differently go-routines.